tasks.py

```diff
@@ -1042,7 +1042,7 @@ def ci_tests(  # noqa: C901 - too complex (9)
     pytest_options = [f"--durations={slowest}", "-rEf"]
 
     if xdist:
-        pytest_options.append("-n auto")
+        pytest_options.append("-n 4")
```

Comment:
Did you mean to hard code this to 4?

Reply:
This was on purpose. Kubernetes exposes all cores of the host machine to the pod even if the pod is allocated fewer

Reply:
Will logical core give the right number?
`4` might be good for CI, but it'll slow things down when testing locally.

Reply:
no, should we make this something that changes when it detects that it's running in CI?